### ISSUE TITLE
Cannot load configuration with log format type Log4jLogFormat

### ISSUE THREAD

**Issue by tomasrollo** [sevdokimov/log-viewer#62]
When trying to start logviewer with Log4jLogFormat specified as log type in the config file I'm getting the below exception.
The exception is not thrown when I use the LogbackLogFormat.

Is this because the Log4jLogFormat class is not specified in LoggerLibSupport.java (there's **null** instead)?
```
public class LoggerLibSupport {

    private static final LoggerLibSupport[] LIB_SUPPORTS = {
            // Logback
            new LoggerLibSupport("logback",
                    "ch.qos.logback.classic.LoggerContext",
                    "com.logviewer.logLibs.logback.LogbackConfigImporter",
                    "com.logviewer.logLibs.logback.LogbackLogFormat"),

            new LoggerLibSupport("log4j",
                    "org.apache.logging.log4j.core.LoggerContext",
                    "com.logviewer.logLibs.log4j.Log4jConfigImporter",
                    null)
    };
```

The exception:
```
2021-06-05_09:45:05.756 [main] WARN  o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'lvLogService': Unsatisfied dependency expressed through field 'accessManager'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'logManager' defined in com.logviewer.LvStandaloneConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.logviewer.services.LvFileAccessManagerImpl]: Factory method 'logManager' threw exception; nested exception is java.lang.IllegalArgumentException: Invalid configuration [line=13]: failed to load the log format
Exception in thread "main" org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'lvLogService': Unsatisfied dependency expressed through field 'accessManager'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'logManager' defined in com.logviewer.LvStandaloneConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.logviewer.services.LvFileAccessManagerImpl]: Factory method 'logManager' threw exception; nested exception is java.lang.IllegalArgumentException: Invalid configuration [line=13]: failed to load the log format
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:586)
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:90)
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:372)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1329)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:572)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:495)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:317)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:315)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:755)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:881)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:551)
        at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:88)
        at com.logviewer.LogViewerMain.startup(LogViewerMain.java:70)
        at com.logviewer.LogViewerMain.main(LogViewerMain.java:234)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'logManager' defined in com.logviewer.LvStandaloneConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.logviewer.services.LvFileAccessManagerImpl]: Factory method 'logManager' threw exception; nested exception is java.lang.IllegalArgumentException: Invalid configuration [line=13]: failed to load the log format
        at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:591)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1245)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1095)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:535)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:495)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:317)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:315)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:251)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1131)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1058)
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:583)
        ... 15 more
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.logviewer.services.LvFileAccessManagerImpl]: Factory method 'logManager' threw exception; nested exception is java.lang.IllegalArgumentException: Invalid configuration [line=13]: failed to load the log format
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
        at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:583)
        ... 27 more
Caused by: java.lang.IllegalArgumentException: Invalid configuration [line=13]: failed to load the log format
        at com.logviewer.impl.LvPatternFormatRecognizer.fromHocon(LvPatternFormatRecognizer.java:64)
        at com.logviewer.LvStandaloneConfig.logManager(LvStandaloneConfig.java:36)
        at com.logviewer.LvStandaloneConfig$$EnhancerBySpringCGLIB$$789ef1e2.CGLIB$logManager$0(<generated>)
        at com.logviewer.LvStandaloneConfig$$EnhancerBySpringCGLIB$$789ef1e2$$FastClassBySpringCGLIB$$5cdab08.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
        at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:365)
        at com.logviewer.LvStandaloneConfig$$EnhancerBySpringCGLIB$$789ef1e2.logManager(<generated>)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
        ... 28 more
Caused by: com.google.gson.JsonParseException: cannot deserialize interface com.logviewer.data2.LogFormat subtype named Log4jLogFormat; did you forget to register a subtype?
        at com.google.gson.typeadapters.RuntimeTypeAdapterFactory$1.read(RuntimeTypeAdapterFactory.java:193)
        at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199)
        at com.google.gson.Gson.fromJson(Gson.java:803)
        at com.google.gson.Gson.fromJson(Gson.java:768)
        at com.google.gson.Gson.fromJson(Gson.java:717)
        at com.google.gson.Gson.fromJson(Gson.java:689)
        at com.logviewer.impl.LvPatternFormatRecognizer.fromHocon(LvPatternFormatRecognizer.java:62)
        ... 39 more
```

My config file (the relevant part):
```
// List of file patterns describing visible files and subdirectories
logs = [
  {
    path: "/logs/**/*.log"  // All files with ".log" extension are visible by default
    format = {
      type: Log4jLogFormat
      pattern: "%date{ISO8601} %-5p %t %c{4}:%L - %m%n"
    }
  }
]
```

**Comment by sevdokimov** [comment]
Thanks for the bug report. This is fixed in the master branch. The fix will be available in the next release. I hope it will be released tomorrow.

**Comment by sevdokimov** [comment]
The fix is https://github.com/sevdokimov/log-viewer/commit/244eee1e05c5e07b8076cfcb813267118e418cc4

**Comment by sevdokimov** [comment]
Fixed in 0.1.7

**Comment by tomasrollo** [comment]
Thanks, just tried it out, works very well with Log4jLogFormat now!
